src/_guides/language/evolution.md

```diff
@@ -180,131 +183,190 @@ turn into edit-time analysis errors.
+  }
+}
+
+void main() {
+  int i = '42'.parseInt();
+  print(i);
+}
 ```
 
-### Dart 2.15
+### Dart 2.6
+_Released 5 November 2019_
+| [Dart 2.6 announcement](https://medium.com/dartlang/dart2native-a76c815e6baf)
 
-Dart 2.15 improved support for function pointers,
-known as _tear-offs._
-In particular, constructor tear-offs are now supported.
-For details, see the [Dart 2.15 announcement][].
+Dart 2.6 introduced a
+[breaking change (dart-lang/sdk#37985)](https://github.com/dart-lang/sdk/issues/37985).
+Inference changes when using Null values in a `FutureOr` context.
+Constraints of the forms similar to `Null <:FutureOr` now yield `Null`
+as the solution for `T`.
 
-[Dart 2.15 announcement]: https://medium.com/dartlang/dart-2-15-7e7a598e508a
+For example: The following code now prints `Null`.
+Before Dart 2.6, it printed `dynamic`.
+The anonymous closure `() {}` returns the `Null` type.
 
-### Dart 2.16
+```dart
+import 'dart:async';
 
-Dart 2.16 didn’t add any features to the Dart language.
+void foo<T>(FutureOr<T> Function() f) { print(T); }
 
-### Dart 2.17
+main() { foo(() {}); }
+```
 
-Dart 2.17 expanded enum functionality with enhanced enums.
-Enhanced enums allow enum declarations to define members 
-including fields, constructors, methods, getters, etc.
-For more information, see the
-[Enhanced enums][] section of the language tour.
+### Dart 2.5
+_Released 10 September 2019_
+| [Dart 2.5 announcement](https://medium.com/dartlang/announcing-dart-2-5-super-charged-development-328822024970)
 
-[Enhanced enums]: /language/enum#declaring-enhanced-enums
+Dart 2.5 didn't add any features to the Dart language, but it did add
+support for [calling native C code][] from Dart code
+using a new **core library, `dart:ffi`.**
 
-Dart 2.17 also added support for super-initializer parameters in constructors.
-Super parameters allow you
-to avoid having to manually pass each parameter
-into the super invocation of a non-redirecting constructor.
-You can instead use super parameters to forward parameters
-to a superclass constructor.
-For more information, 
-see the language tour's documentation of [super parameters][].
+### Dart 2.4
+_Released 27 June 2019_
 
-[super parameters]: /language/constructors#super-parameters
 
-Dart 2.17 also removed some restrictions on named arguments.
-Named arguments can now be freely interleaved with positional arguments.
-The following code was invalid before 2.17,
-but is now allowed:
+Dart 2.4 introduces a breaking change
+[dart-lang/sdk#35097](https://github.com/dart-lang/sdk/issues/35097).
+
+Dart now enforces covariance of type variables used in super-interfaces.
+For example: Prior to this release Dart accepted, but now rejects,
+the following code:
 
 ```dart
-void main() {
-  test(skip: true, 'A test description', () {
-    // Very long function body here...
-  });
+class A<X> {};
+class B<X> extends A<void Function(X)> {};
+```
+
+You can now use the identifier async in asynchronous and generator functions.
```

Comment:
Something doesn't exist as an identifier until you use it as such, so the following may read better:

```suggestion
You can now use `async` as an identifier in asynchronous and generator functions.
```